src/utils/metadata/parser.rs

```diff
@@ -95,6 +95,12 @@ impl ParserMetadata {
         global_id
     }
 
+    pub fn is_shadowing_prev_scope(&self, name: &str) -> bool {
+        self.context.scopes.len() > 1
+            && self.context.scopes.iter().rev().skip(1)
```

Comment:
I try to separate function call chains out onto their own lines, to make it easier to see what's going on:
```
pub fn is_shadowing_prev_scope(&self, name: &str) -> bool {
    if self.context.scopes.len() > 1 {
        self.context.scopes.iter()
            .rev()
            .skip(1)
            .find_map(|scope| scope.get_var(name))
            .is_some()
    } else {
        false
    }
}
```